### PR TITLE
Fix multiple verify calls and Add connect wallet button

### PR DESCRIPTION
**Pull request by yurikoinaba**
connect wallet button
dont allow multiple verify calls by disabling all buttons during verification
connect wallet modal appears on zkverify button click if not connected already

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -15,9 +14,9 @@
     "@chakra-ui/react": "^2.3.6",
     "@emotion/react": "^11.10.5",
     "@emotion/styled": "^11.10.5",
-    "@thirdweb-dev/react": "^3.4.4",
-    "@thirdweb-dev/sdk": "^3.4.4",
+    "@talismn/connect-components": "^1.1.8",
     "@types/seedrandom": "^3.0.2",
+    "antd": "^5.21.2",
```
Comment (yurikoinaba) [inline package.json]: This is fine for this purpose, but the components are probably also available in `@chakra-ui/react` which is already installed

File: context/useZkVerify.ts [hunk 2]
```diff
@@ -25,51 +29,56 @@ export function useZkVerify(selectedAccount: string | null) {
...
 
       let session;
       try {
         session = await zkVerifySession.start().Testnet().withWallet();
       } catch (error: unknown) {
+        if ((error as Error).message.includes('User rejected the transaction')) {
+          localCancelled = true;
+          setCancelled(true);
+          return { verified: false, cancelled: true };
+        }
```
Comment (yurikoinaba) [inline context/useZkVerify.ts]: How is the rejected transaction related to starting a zkVerify session?

File: context/useZkVerify.ts [hunk 3]
```diff
@@ -25,51 +29,56 @@ export function useZkVerify(selectedAccount: string | null) {
...
       });
 
       let transactionInfo = null;
       try {
         transactionInfo = await transactionResult;
       } catch (error: unknown) {
+        if ((error as Error).message.includes('Rejected by user')) {
+          localCancelled = true;
+          setCancelled(true);
+          return { verified: false, cancelled: true };
```
Comment (yurikoinaba) [inline context/useZkVerify.ts:38]: Returning the object here and not relying on the states defeats the purpose of storing the values as a state

Comment (yurikoinaba) [reply context/useZkVerify.ts:38]: In the context of this game where multiple proofs can be submitted, it doesn't make sense to have this zkVerify utility as a hook, and the `verifying`, `verified`, `error`, and `cancelled` states are redundant and can be removed.

Handling those states are saved per row in `GameContext`

File: context/useZkVerify.ts [hunk 4]
```diff
@@ -25,51 +29,56 @@ export function useZkVerify(selectedAccount: string | null) {
...
+        return { verified: false, cancelled: false, error: errorMessage };
+      }
+      return { verified: false, cancelled: true };
     } finally {
       setVerifying(false);
     }
   };
 
-  return { verifying, verified, error, onVerifyProof };
+  return { onVerifyProof, verifying, verified, error, cancelled };
```
Comment (yurikoinaba) [inline context/useZkVerify.ts]: So this can just be `return { onVerifyProof };`

File: context/useZkVerify.ts [hunk 5]
```diff
@@ -4,11 +4,15 @@ export function useZkVerify(selectedAccount: string | null) {
   const [verifying, setVerifying] = useState(false);
```
Comment (yurikoinaba) [inline context/useZkVerify.ts]: The `selectedAccount` parameter isn't being used

**Comment by yurikoinaba** [comment]
- I'm getting type errors on `npm run build`

Lower priority bugs:
- I chose to connect w Talisman, but when I click the button to "zkVerify" the SubWallet popup appears
- UI is already showing a connected state before I select my account to connect
Screen Shot 2024-10-07 at 12 19 22 PM

File: context/GameContext.tsx [hunk 6]
```diff
@@ -199,22 +166,28 @@ const gameReducer = (state: Game, action: GameAction) => {
...
 const GameProvider: React.FC<{ children: JSX.Element }> = ({ children }) => {
   const toast = useToast();
-
-  const [game, dispatch] = React.useReducer(
-    gameReducer,
-    JSON.parse(JSON.stringify(DEFAULT_GAME))
-  );
-  const { verifying, verified, error, onVerifyProof } = useZkVerify(null);
+  const [game, dispatch] = React.useReducer(gameReducer, JSON.parse(JSON.stringify(DEFAULT_GAME)));
+  const [verifying, setVerifying] = useState(false);
```
Comment (yurikoinaba) [inline context/GameContext.tsx:188]: One improvement would be to keep track of the row that's being verified. 
`const [verifying, setVerifying] = useState('');`
Then in the UI, only one button will be in loading state and the others can be just in the disabled state. Not a blocker.

Screen Shot 2024-10-08 at 12 25 01 PM

**Review by yurikoinaba — COMMENTED** [review]
lgtm

**Review by rushby — APPROVED** [review]
Yuriko helped rebase my changes, and has reviewed everything.

**Pull request merged**